src/sentry/api/event_search.py

```diff
@@ -2015,13 +2015,18 @@ def resolve_field(field, params=None):
     return ([field], None)
 
 
-def resolve_field_list(fields, snuba_filter, auto_fields=True):
+def resolve_field_list(fields, snuba_filter, auto_fields=True, auto_aggregations=None):
     """
     Expand a list of fields based on aliases and aggregate functions.
 
     Returns a dist of aggregations, selected_columns, and
     groupby that can be merged into the result of get_snuba_query_args()
     to build a more complete snuba query based on event search conventions.
+
+    Auto aggregates are aggregates that will be automatically added to the
+    list of aggregations when they're used in a condition. This is so that
+    they can be used in a condition without having to manually add the
+    aggregate to a field.
```

Comment:
As discussed in slack, I'll enable this on both for all aggregates when there's at least one aggregate so that we aren't changing grouping on users in unexpected ways.
:+1: